tiling.js

```diff
@@ -1515,13 +1522,164 @@ class Spaces extends Map {
                 continue;
             monitor.clickOverlay.activate();
         }
+
+        inPreviewMode = {stack: false, sequence: false};
+    }
+
+    _initWorkspaceSequence() {
```

Comment:
not sure what this one is, I do see animation completion in your commit is that what you meant

Reply:
Yeah, I made quick hack that calls out to `_initWorkspaceSequence` in `animateToSpace`, but that requires resetting `inPreview` so not very pretty. Meaning we probably just want a shared method that sets up spaces in sequential order which both `_initWorkspaceSequence` and `animateToSpace` can use. Does that make sense?

Reply:
got it, will work on that

Reply:
So I looked over animateToSpace and switchToWorkspace and from what I see _initWorkspaceSequence is called before animateToSpace occurs and from what I can tell it's sufficient and have not had any side effects I can see after testing. I did put a conditional around some code in animateToSpace specific to stack arrangement but that was minor. please see 903dea3

also I removed a tween from _initWorkspaceSequence it seemed unneed to me (I could be wrong) but _initWorkspaceSequence also does not have it. Not sure what use case you were thinking of there, but in my testing it made no difference. See cfef02e

Reply:
Currently `animateToSpace` only handles the selected space correctly when finishing a sequential animation. You can often see the space below overlapping the selected space.
![image](https://user-images.githubusercontent.com/71978/68542260-25ff4a00-03ab-11ea-8ec6-ee35ca81d028.png)

This is fixable in two ways:
- Make sure the current space is raised above the other spaces
- Just animate all the whole sequence of spaces to its destination, this happens to involve the same code as `_initWorkspaceSequence`, but with animation. Meaning it makes sense to pull that out into its own method, and use it to both set up the sequence and finish the selection.

 Now I have some local fixes, so I'll push those to avoid any duplicate work.